mlflow/gateway/client.py

```diff
@@ -116,3 +116,52 @@ def search_routes(self, search_filter: Optional[str] = None):
             )
         response = self._call_endpoint("GET", self._route_base).json()["routes"]
         return [Route(**resp) for resp in response]
+
+    def query(self, route: str, data: Dict[str, Any]):
+        """
+        Submit a query to a configured provider route.
+
+        :param route: The name of the route to submit the query to.
+        :param data: The data to send in the query. A dictionary representing the per-route
+        specific structure required for a given provider.
+
+        For chat, the structure should be:
+
+        .. code-block:: python
+          from mlflow.gateway import MlflowGatewayClient
+
+          gateway_client = MlflowGatewayClient("http://my.gateway:8888")
+
+          response = gateway_client.query(
+              "my-chat-route",
+              {"messages": [{"role": "user", "content": "Tell me a joke about rabbits"}, ...]},
+          )
+
+        For completions, the structure should be:
+
+        .. code-block:: python
+          from mlflow.gateway import MlflowGatewayClient
+
+          gateway_client = MlflowGatewayClient("http://my.gateway:8888")
+
+          response = gateway_client.query("my-chat-route", {"prompt": "It's one small step for"})
+
+        For embeddings, the structure should be:
+
+        .. code-block:: python
+          from mlflow.gateway import MlflowGatewayClient
+
+          gateway_client = MlflowGatewayClient("http://my.gateway:8888")
+
+          response = gateway_client.query(
+              "my-chat-route", {"text": ["It was the best of times", "It was the worst of times"]}
+          )
```

Comment:
```suggestion
          response = gateway_client.query("my-completions-route", {"prompt": "It's one small step for"})

        For embeddings, the structure should be:

        .. code-block:: python
          from mlflow.gateway import MlflowGatewayClient

          gateway_client = MlflowGatewayClient("http://my.gateway:8888")

          response = gateway_client.query(
              "my-embeddings-route", {"text": ["It was the best of times", "It was the worst of times"]}
          )
```

(love the example prompts / text, btw)

Reply:
@dbczumar Updated!

Reply:
Thanks Haru! Let's merge! :D

Reply:
Should be auto-merged soon once CI passes